### ISSUE TITLE
CMake move include(GNUInstallDirs) below project

### ISSUE THREAD

**Issue by timrulebosch** [redis/hiredis#1022]
The include for GNUInstallDirs should come after the project(). Like this:

```
CMAKE_MINIMUM_REQUIRED(VERSION 3.4.0)
PROJECT(hiredis)
INCLUDE(GNUInstallDirs)
```

Otherwise, we get warnings from newer versions of CMake.

```
CMake Warning (dev) at /snap/cmake/992/share/cmake-3.22/Modules/GNUInstallDirs.cmake:239 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:2 (INCLUDE)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

https://github.com/redis/hiredis/blob/b5716ee82926316f7764b834eec636f5652d5600/CMakeLists.txt#L2

I can provide a PR if you like.

**Comment by bjosv** [comment]
The PR #994 touches this, but for a different reason

**Comment by bjosv** [comment]
This issue can be closed now, fixed via the merged #872

**Comment by michael-grunder** [comment]
Closing via #872
